clpy/core/core.pyx

```diff
@@ -4247,41 +4247,43 @@ def _inclusive_scan_kernel(dtype, block_size):
     name = "inclusive_scan_kernel"
     dtype = _get_typename(dtype)
     source = string.Template("""
-    extern "C" __global__ void ${name}(const CArray<${dtype}, 1> src,
-        CArray<${dtype}, 1> dst){
-        long long n = src.size();
-        extern __shared__ ${dtype} temp[];
-        unsigned int thid = threadIdx.x;
-        unsigned int block = 2 * blockIdx.x * blockDim.x;
-
-        unsigned int idx0 = thid + block;
-        unsigned int idx1 = thid + blockDim.x + block;
+    __kernel void ${name}(
+            const CArray<${dtype}, 1> src,
+            CArray<${dtype}, 1> dst
+            ){
+        size_t n = src.size();
+        __local ${dtype} temp[${block_size}*2];
+        size_t thid = get_local_id(0);
+        size_t block = 2 * get_group_id(0) * get_local_size(0);
```

Comment:
I know that you expect original CUDA code to minimize amount of modifications.
However, it confuses me because this is OpenCL C kernel therefore there is not `block`.

Can you rewrite its and others' name in OpenCL C terminology?

Reply:
Actually, these `block`s and `block_size`s don't share the meaning with CUDA.
Anyway, I tried to avoid saying `block` in the scan kernel. I've had been even confused.